File: 111. Minimum Depth of Binary Tree/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        int depth = 0;
        queue<TreeNode*> q;
        if (root != nullptr) q.push(root);
        while (!q.empty()) {
            int l = q.size();
            depth++;
            for (int i = 0; i < l; i++) {
                TreeNode* curr = q.front();
                q.pop();
                if (curr -> left == nullptr && curr -> right == nullptr) return depth;
                if (curr -> left != nullptr) q.push(curr -> left);
                if (curr -> right != nullptr) q.push(curr -> right);
            }
        }
        return depth;
    }
};

File: 111. Minimum Depth of Binary Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        depth = 0
        q = [root] if root else []
        while q:
            depth += 1
            l = len(q)
            for i in range(l):
                curr = q.pop(0)
                if not curr.left and not curr.right: return depth
                if curr.left: q.append(curr.left)
                if curr.right: q.append(curr.right)
        return depth
